# Лабораторная работа 3.2. Домашнее задание

### Задание №1

Создайте два массива: в первом должны быть четные числа от 2 до 12 включительно, а в другом числа 7, 11, 15, 18, 23, 29.

$1.$ Сложите массивы и возведите элементы получившегося массива в квадрат:

In [1]:
import numpy as np


a = np.arange(2, 8)
b = np.array([7, 11, 15, 18, 23, 29])

print("Элементы: ", np.square(a+b))

Элементы:  [  81  196  361  529  841 1296]


$2.$ Выведите все элементы из первого массива, индексы которых соответствуют индексам тех элементов второго массива, которые больше 12 и дают остаток 3 при делении на 5.

In [2]:
import numpy as np

a = np.arange(2, 8)
b = np.array([7, 11, 15, 18, 23, 29])

print(a[np.logical_and(b > 12, b % 5 == 3)])

[5 6]


*3.* Проверьте условие "Элементы первого массива делятся на 4, элементы второго массива меньше 14". (Подсказка: в результате должен получиться массив с True и False)

In [2]:
import numpy as np

a = np.arange(2, 8)
b = np.array([7, 11, 15, 18, 23, 29])

print(a[np.logical_and(b > 12, b % 5 == 3)])
print(a % 4 == 0)
print(b < 14)

[5 6]
[False False  True False False False]
[ True  True False False False False]


### Задание №2

* Найдите интересный для вас датасет. Например, можно выбрать датасет тут. http://data.un.org/Explorer.aspx (выбираете датасет, жмете на view data, потом download, выбирайте csv формат)
* Рассчитайте подходящие описательные статистики для признаков объектов в выбранном датасете
* Проанализируйте и прокомментируйте содержательно получившиеся результаты
* Все комментарии оформляйте строго в ячейках формата markdown
    

In [69]:
import csv
import numpy as np

with open('economy.csv', 'r', newline='') as csvfile:
    data = csv.reader(csvfile, delimiter=',')
    data_value = []
    for row in data:
        if row[1] == "Marine services":
            data_value.append(float(row[6]))
    operation = len(data_value)
    print("Количество операций в сфере морских услуг: "
        f"{operation}")
    

Количество операций в сфере морских услуг: 92


Расчитаем описательную статистику. Приведём список к надлежащему типу.

In [65]:
data_value = np.array(data_value)
type(data_value)

numpy.ndarray

In [75]:
print("Количество денежных средств заработанных на морских услугах: "
        f"{round(np.sum(data_value))}$")

Количество денежных средств заработанных на морских услугах: 2957331$


In [78]:
print("Минимальный доход (расход) в сфере морских услуг: "
        f"{round(np.min(data_value))}$")

Минимальный доход (расход) в сфере морских услуг: -6739$


In [79]:
print("Максимальный доход в сфере морских услуг: "
        f"{round(np.max(data_value))}$")

Максимальный доход в сфере морских услуг: 234404$


In [81]:
print("Средний доход в сфере морских услуг: "
        f"{round(np.mean(data_value))}$")

Средний доход в сфере морских услуг: 32145$


In [85]:
print("Стандартное отклонение: "
        f"{round(np.std(data_value))}$")

Стандартное отклонение: 57675$


In [86]:
print("Дисперсия: "
        f"{round(np.var(data_value))}$")

Дисперсия: 3326411652$


В результате мы можем сказать, что морских услуг было реализовано 98. За них было получено 2957331. Оплата минимальной услуги составила -6739, что свидетельствует о работе в убыток. Максимальная оплата морской услуги составила 234404. Средних доход за реализацию морских услуг 32145. Стандартное отклонение - 57675, свидетельствует о большой разнице между доходами (элементами массива). Дисперсия составила 3326411652, что свидетельствует о большом разбросе доходов (элементов массива) от его среднего значения. 